test_group_chain fails with python-click 6.7+git20180829-1

As reported by Adrian Bunk in [Debian Bug #908684](https://bugs.debian.org/908684), the python-click-plugins package fails to build with python-click 6.7+git20180829-1 due to a test failure:

```
I: pybuild pybuild:269: cp -r /build/python-click-plugins-1.0.2/tests /build/python-click-plugins-1.0.2/.pybuild/cpython2_2.7_click-plugins/build
I: pybuild base:217: cd /build/python-click-plugins-1.0.2/.pybuild/cpython2_2.7_click-plugins/build; python2.7 -m pytest tests
============================= test session starts ==============================
platform linux2 -- Python 2.7.15+, pytest-3.6.4, py-1.6.0, pluggy-0.6.0
rootdir: /build/python-click-plugins-1.0.2, inifile:
collected 5 items

tests/test_plugins.py ...F.                                              [100%]

=================================== FAILURES ===================================
_______________________________ test_group_chain _______________________________

runner = <click.testing.CliRunner object at 0x7f9d888d9050>

    def test_group_chain(runner):
    
        # Attach a sub-group to a CLI and get execute it without arguments to make
        # sure both the sub-group and all the parent group's commands are present
        @good_cli.group()
        def sub_cli():
            """Sub CLI."""
            pass
    
        result = runner.invoke(good_cli)
        assert result.exit_code is 0
        assert sub_cli.name in result.output
        for ep in iter_entry_points('_test_click_plugins.test_plugins'):
            assert ep.name in result.output
    
        # Same as above but the sub-group has plugins
        @with_plugins(plugins=iter_entry_points('_test_click_plugins.test_plugins'))
        @good_cli.group()
        def sub_cli_plugins():
            """Sub CLI with plugins."""
            pass
    
        result = runner.invoke(good_cli, ['sub_cli_plugins'])
>       assert result.exit_code is 0
E       assert 2 is 0
E        +  where 2 = <Result SystemExit(2,)>.exit_code

tests/test_plugins.py:124: AssertionError
====================== 1 failed, 4 passed in 0.12 seconds ======================
E: pybuild pybuild:338: test: plugin distutils failed with: exit code=1: cd /build/python-click-plugins-1.0.2/.pybuild/cpython2_2.7_click-plugins/build; python2.7 -m pytest tests
```

The python-click package was recently updated to a git snapshot which likely changed the behaviour.

--- a/.gitignore
+++ b/.gitignore
@@ -1,29 +1,29 @@
 # Byte-compiled / optimized / DLL files
 __pycache__/
 *.py[cod]
+*$py.class
 
 # C extensions
 *.so
 
 # Distribution / packaging
 .Python
-env/
-venv/
-venv2/
-venv3/
 build/
 develop-eggs/
 dist/
 downloads/
 eggs/
+.eggs/
 lib/
 lib64/
 parts/
 sdist/
 var/
+wheels/
 *.egg-info/
 .installed.cfg
 *.egg
+MANIFEST
 
 # PyInstaller
 #  Usually these files are written by a python script from a template
@@ -38,10 +38,15 @@ pip-delete-this-directory.txt
 # Unit test / coverage reports
 htmlcov/
 .tox/
+.nox/
 .coverage
+.coverage.*
 .cache
 nosetests.xml
 coverage.xml
+*.cover
+.hypothesis/
+.pytest_cache/
 
 # Translations
 *.mo
@@ -49,6 +54,15 @@ coverage.xml
 
 # Django stuff:
 *.log
+local_settings.py
+db.sqlite3
+
+# Flask stuff:
+instance/
+.webassets-cache
+
+# Scrapy stuff:
+.scrapy
 
 # Sphinx documentation
 docs/_build/
@@ -56,5 +70,42 @@ docs/_build/
 # PyBuilder
 target/
 
-# Intellij PyCharm
-.idea/
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+.python-version
+
+# celery beat schedule file
+celerybeat-schedule
+
+# SageMath parsed files
+*.sage.py
+
+# Environments
+.env
+.venv
+env/
+venv/
+ENV/
+env.bak/
+venv.bak/
+
+# Spyder project settings
+.spyderproject
+.spyproject
+
+# Rope project settings
+.ropeproject
+
+# mkdocs documentation
+/site
+
+# mypy
+.mypy_cache/
+.dmypy.json
+dmypy.json

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,9 +7,9 @@ cache:
 
 python:
   - 2.7
-  - 3.3
   - 3.4
   - 3.5
+  - 3.6
   - pypy
   - pypy3
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,7 +18,7 @@ install:
   - pip install -e .\[dev\]
 
 script:
-  - py.test tests --cov click_plugins --cov-report term-missing
+  - pytest tests --cov click_plugins --cov-report term-missing
 
 after_success:
   - coveralls

--- a/README.rst
+++ b/README.rst
@@ -165,7 +165,7 @@ Developing
     $ git clone https://github.com/click-contrib/click-plugins.git
     $ cd click-plugins
     $ pip install -e .\[dev\]
-    $ py.test tests --cov click_plugins --cov-report term-missing
+    $ pytest tests --cov click_plugins --cov-report term-missing
 
 
 Changelog

--- a/setup.py
+++ b/setup.py
@@ -51,7 +51,7 @@ setup(
                 "via setuptools entry-points.",
     extras_require={
         'dev': [
-            'pytest',
+            'pytest>=3',
             'pytest-cov',
             'wheel',
             'coveralls'

--- a/tests/test_plugins.py
+++ b/tests/test_plugins.py
@@ -115,18 +115,18 @@ def test_group_chain(runner):
 
     # Same as above but the sub-group has plugins
     @with_plugins(plugins=iter_entry_points('_test_click_plugins.test_plugins'))
-    @good_cli.group()
+    @good_cli.group(name='sub-cli-plugins')
     def sub_cli_plugins():
         """Sub CLI with plugins."""
         pass
 
-    result = runner.invoke(good_cli, ['sub_cli_plugins'])
+    result = runner.invoke(good_cli, ['sub-cli-plugins'])
     assert result.exit_code is 0
     for ep in iter_entry_points('_test_click_plugins.test_plugins'):
         assert ep.name in result.output
 
     # Execute one of the sub-group's commands
-    result = runner.invoke(good_cli, ['sub_cli_plugins', 'cmd1', 'something'])
+    result = runner.invoke(good_cli, ['sub-cli-plugins', 'cmd1', 'something'])
     assert result.exit_code is 0
     assert result.output.strip() == 'passed'
 

